src/widgetastic/widget.py

```diff
@@ -1322,7 +1322,7 @@ class TableRow(Widget, ClickableMixin):
 
     def __init__(self, parent, index, logger=None):
         Widget.__init__(self, parent, logger=logger)
-        self.index = index
+        self.index = index + 1 if parent._is_header_in_body else index or 1
```

Comment:
if an index of 0 was passed erroneously when the header is separate, default to an index of 1 instead since this is formatted into an XPATH.

Reply:
*OPTIONAL*
I'm kind of wondering if `Table` should always be responsible for passing in the right index. In other words we don't do any of the "math" here and leave `Table` to be in charge of setting proper index when creating the `TableRow`. That way the math is all contained in 1 place and you know exactly what your index is going to be when you see `TableRow` instantiated in `Table`. You've probably already thought about that so let me know ... ;)

Reply:
Will check out all the other comments and come back to this.